expr/expr_test.go

```diff
@@ -2081,6 +2081,35 @@ func TestEvalExpression(t *testing.T) {
 			[]*MetricData{makeResponse("removeAbovePercentile(metric1, 50)",
 				[]float64{1, 2, -1, 7, math.NaN(), math.NaN(), math.NaN(), math.NaN()}, 1, now32)},
 		},
+		{
+			&expr{
+				target: "cactiStyle",
+				etype:  etFunc,
+				args: []*expr{
+					{target: "metric1"},
+					{valStr: "si", etype: etString},
+				},
+			},
+			map[MetricRequest][]*MetricData{
+				MetricRequest{"metric1", 0, 1}: {makeResponse("metric1", []float64{1, 2, 3, 4, 5, math.NaN()}, 1, now32)},
+			},
+			[]*MetricData{makeResponse("metric1 Current: 5 Max: 5 Min: 1", []float64{1, 2, 3, 4, 5, math.NaN()}, 1, now32)},
+		},
+		{
+			&expr{
+				target: "cactiStyle",
+				etype:  etFunc,
+				args: []*expr{
+					{target: "metric1"},
+					{valStr: "si", etype: etString},
+					{valStr: "carrots", etype: etString},
+				},
+			},
+			map[MetricRequest][]*MetricData{
+				MetricRequest{"metric1", 0, 1}: {makeResponse("metric1", []float64{1, 2, 3, 4, 5}, 1, now32)},
+			},
+			[]*MetricData{makeResponse("metric1 Current: 5 carrots Max: 5 carrots Min: 1 carrots", []float64{1, 2, 3, 4, 5}, 1, now32)},
+		},
```

Comment:
Suggestions for additional test cases:

1. 1/3
1. 0 +/- e (some very small value)
1. all math.NaN()
1. 1000 exactly
1. slightly more than each of: 10^6, 10^9, 10^12, 10^15
1. similarly: 10^-3, 10^-6, ...